app/workers/web_push_notification_worker.rb

```diff
@@ -5,23 +5,18 @@ class WebPushNotificationWorker
 
   sidekiq_options backtrace: true
 
-  def perform(recipient_id, notification_id)
-    recipient = Account.find(recipient_id)
+  def perform(session_activation_id, notification_id)
+    session_activation = SessionActivation.find(session_activation_id)
     notification = Notification.find(notification_id)
 
-    sessions_with_subscriptions = recipient.user.session_activations.reject { |session| session.web_push_subscription.nil? }
+    begin
+      session_activation.web_push_subscription.push(notification)
+    rescue Webpush::InvalidSubscription, Webpush::ExpiredSubscription => e
+      # Subscription expiration is not currently implemented in any browser
+      session_activation.web_push_subscription.destroy!
+      session_activation.update!(web_push_subscription: nil)
 
-    sessions_with_subscriptions.each do |session|
-      begin
-        session.web_push_subscription.push(notification)
-      rescue Webpush::InvalidSubscription, Webpush::ExpiredSubscription
-        # Subscription expiration is not currently implemented in any browser
-        session.web_push_subscription.destroy!
-        session.web_push_subscription = nil
-        session.save!
-      rescue Webpush::PayloadTooLarge => e
-        Rails.logger.error(e)
-      end
+      throw e
```

Comment:
It is possible to `throw`/`catch` in Ruby but it's a different way of doing control flow... Not sure what is intended in this case. (<https://stackoverflow.com/questions/51021/what-is-the-difference-between-raising-exceptions-vs-throwing-exceptions-in-ruby>)